caas/kubernetes/provider/bootstrap.go

```diff
@@ -554,52 +559,149 @@ func (c *controllerStack) createControllerStatefulset() error {
 	if err = c.broker.createStatefulSet(spec); err != nil {
 		return errors.Trace(err)
 	}
-	if err = c.syncPodStatus(w); err != nil {
-		return errors.Annotate(err, "fetching pods' status for controller")
+
+	for i := int32(0); i < numberOfPods; i++ {
+		podName := c.pcfg.GetPodName() // TODO(caas): HA mode!
+		if err = c.waitForPod(w, podName); err != nil {
+			return errors.Trace(err)
+		}
 	}
 	return nil
 }
 
-func (c *controllerStack) syncPodStatus(w watcher.NotifyWatcher) error {
+func (c *controllerStack) waitForPod(podWatcher watcher.NotifyWatcher, podName string) error {
+	timeout := c.broker.clock.NewTimer(c.pcfg.Bootstrap.Timeout)
+
+	podEventWatcher, err := c.broker.watchEvents(podName, "Pod")
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer podEventWatcher.Kill()
+
 	printedMsg := set.NewStrings()
-	checkContainerEvents := func(events []core.Event, reason string, eventCount int) bool {
-		count := 0
+	printPodEvents := func() error {
+		events, err := c.broker.getEvents(podName, "Pod")
+		if err != nil {
+			return errors.Trace(err)
+		}
 		for _, evt := range events {
+			logger.Debugf("%s %v - %v", podName, evt.Reason, evt.Message)
 			// clean the messages to prevent duplicated records.
 			// we don't care which image is been pulling/pulled and this reason should be printed once only.
 			switch evt.Reason {
 			case PullingImage:
 				evt.Message = "Downloading images"
 			case PulledImage:
 				evt.Message = "Pulled images"
-			case reason:
-				count++
+			case StartedContainer:
+				if evt.InvolvedObject.FieldPath == fmt.Sprintf("spec.containers{%s}", mongoDBContainerName) {
+					evt.Message = "Started mongodb container"
```

Comment:
yeah, so we don't wanna show which container is started,

```text
Creating Juju controller "k1" on microk8s/localhost
Creating k8s resources for controller "controller-k1"
Downloading images
Starting controller pod
```
just show `Starting controller pod` once all the `two` images have been `Pulled`